### PR TITLE
Fix typos and clarify phrases

### PR DESCRIPTION
**Pull request by advancedxy**
see file changes for more detail.

### CODE REVIEW

File: manuscript/08-Iterators-And-Generators.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Iterators are nothing more than objects with a certain interface. That interface
 
 If you call `next()` after the last value has been returned, the method returns `done` as `true` and `value` contains the return value for the iterator. The *return value* is not considered part of the data set, but rather a final piece of related data or `undefined` if no such data exists. (This concept will become clearer in the generators section later in this chapter.)
 
-With that understanding, it's fairly easy to create an iterator using ECMAScript 5, for example:
+With that understanding, it's fairly easy to create an iterator using ECMAScript 6, for example:
```
Comment (nzakas) [inline manuscript/08-Iterators-And-Generators.md]: Incorrect. The example uses only es5.

**Comment by nzakas** [comment]
Sorry, neither change make sense to me.

File: manuscript/08-Iterators-And-Generators.md (hunk 1 above)
Comment (advancedxy) [inline manuscript/08-Iterators-And-Generators.md]: It's weird to give an es5 example when you are writing about es6. But I won't argue with you for that.

File: manuscript/08-Iterators-And-Generators.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ console.log(iterator.next());           // "{ value: undefined, done: true }"
 console.log(iterator.next());           // "{ value: undefined, done: true }"
 ```
 
-The `createIterator()` function in this example returns an object with a `next()` method. Each time the method is called, the next value in the `items` array is returned as `value`. When `i` is 4, `items[i++]` returns `undefined` and `done` is `true`, which fulfills the special last case for iterators in ECMAScript 6.
+The `createIterator()` function in this example returns an object with a `next()` method. Each time the method is called, the next value in the `items` array is returned as `value`. When `i` is 4, `value` returns `undefined` and `done` is `true`, which fulfills the special last case for iterators in ECMAScript 6.
```
Comment (advancedxy) [inline manuscript/08-Iterators-And-Generators.md]: It's `value` that returns not the `items[i++]`.

File: manuscript/08-Iterators-And-Generators.md (hunk 1 above)
Comment (nzakas) [inline manuscript/08-Iterators-And-Generators.md]: The point here is that there's no magic in iterators, it's just a description of how something works, and so it's easy to create it yourself even in es5.

File: manuscript/08-Iterators-And-Generators.md (hunk 2 above)
Comment (nzakas) [inline manuscript/08-Iterators-And-Generators.md]: Value is a variable, it can't return anything. Items[I++] is an operation, so it can return something.

**Pull request closed without merging**